compendium/test_bundles/TestBundleDSTOI18/src/ServiceImpl.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef _SERVICE_IMPL_HPP_
+#define _SERVICE_IMPL_HPP_
+
+#include "cppmicroservices/servicecomponent/ComponentContext.hpp"
+#include "TestInterfaces/Interfaces.hpp"
+
+using ComponentContext = cppmicroservices::service::component::ComponentContext;
+
+namespace sample
+{
+  class ServiceComponent18 : public test::Interface3
```

Comment:
its a good habit to mark `class ServiceComponent18` as `final` to stop subclassing of `ServiceComponent18`